src/main/java/green/projectile/Projectile.java

```diff
@@ -37,6 +41,23 @@ public double getInterceptX() {
         return getHorizontalVelocity() * getTimeAtXInt();
     }
 
+    public double getTimeAtX(double x) {
+        return x / getHorizontalVelocity();
+    }
+
+    public double getTimeAtY(double y) {
+        return y / getVerticalVelocity();
+    }
+
+    public double getYAtX(double x) {
+        double seconds = getTimeAtX(x);
+        return Math.sin(angle) * velocity * seconds - (HALF * GRAVITY * seconds * seconds);
+    }
+
+    public double getXAtY(double y) {
+        return getHorizontalVelocity() * getTimeAtY(y);
+    }
```

Comment:
I have doubts about this. Technically speaking there are two possible x points for every y.